### PR TITLE
🌱 Bump CAPI to v1.7.0-beta.0

### PR DESCRIPTION
**Pull request by kashifest**
Bumps CAPI to v1.7.0-beta.0 and prepare main branch for CAPI v1.7 contract.

### CODE REVIEW

**Comment by kashifest** [comment]
Note for reviewers: This has an extra commit which will be removed once #1525 lands

**Comment by kashifest** [comment]
Feature test Passed here: https://jenkins.nordix.org/job/metal3_capm3_main_e2e_feature_test_ubuntu/111/

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

File: scripts/ci-e2e.sh [hunk 1]
```diff
@@ -29,8 +29,8 @@ source "${REPO_ROOT}/scripts/environment.sh"
 # Clone dev-env repo
 sudo mkdir -p ${WORKING_DIR}
 sudo chown "${USER}":"${USER}" ${WORKING_DIR}
-M3_DEV_ENV_REPO="https://github.com/metal3-io/metal3-dev-env.git"
-M3_DEV_ENV_BRANCH=main
+M3_DEV_ENV_REPO="https://github.com/Nordix/metal3-dev-env.git"
+M3_DEV_ENV_BRANCH=bump/capi-v1.7.0beta
```
Comment (mboukhalfa) [inline scripts/ci-e2e.sh:33]: shouldnt we merge that to main first it looks weird to merge this to main than revert it

Comment (kashifest) [reply scripts/ci-e2e.sh]: Check the commit message

Comment (kashifest) [reply scripts/ci-e2e.sh]: Removed now.

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Pull request merged**